src/Runner.Common/JobServerQueue.cs

```diff
@@ -292,14 +306,46 @@ private async Task ProcessWebConsoleLinesQueueAsync(bool runOnce = false)
                         {
                             try
                             {
-                                // we will not requeue failed batch, since the web console lines are time sensitive.
-                                if (batch[0].LineNumber.HasValue)
+                                bool pushedLinesViaWebsocket = false;
+                                if (this._websocketClient?.State == WebSocketState.Open)
                                 {
-                                    await _jobServer.AppendTimelineRecordFeedAsync(_scopeIdentifier, _hubName, _planId, _jobTimelineId, _jobTimelineRecordId, stepRecordId, batch.Select(logLine => logLine.Line).ToList(), batch[0].LineNumber.Value, default(CancellationToken));
+                                    var jsonData = JsonConvert.SerializeObject(new
+                                    {
+                                        stepRecordId,
+                                        startLine = batch[0].LineNumber,
+                                        lines = batch.Select(logLine => logLine.Line).ToList()
+                                    });
+                                    Trace.Info($"Sending to websocket: {jsonData}");
+                                    try
+                                    {
+                                        // It should be okay to wait for the result since we are already doing this in the background and doing it one by one
+                                        await this._websocketClient.SendAsync(new ArraySegment<byte>(Encoding.UTF8.GetBytes(jsonData)), WebSocketMessageType.Text, false, default(CancellationToken));
+                                        pushedLinesViaWebsocket = true;
+                                    }
+                                    catch (Exception ex)
+                                    {
+                                        Trace.Info("Catch exception during append web console line to websocket, let's fallback to sending via non-websocket call.");
```

Comment:
@TingluoHuang : may be let's also short circuit when things bail out, give some max attempts